packages/@lwc/synthetic-shadow/src/faux-shadow/events.ts

```diff
@@ -384,18 +441,24 @@ export function addShadowRootEventListener(
             );
         }
     }
-    const elm = getHost(sr);
-    const wrappedListener = getWrappedShadowRootListener(sr, listener);
-    attachDOMListener(elm, type, wrappedListener);
+    // TODO [#1824]: Lift this restriction on the option parameter
+    if (isFunction(listener)) {
+        const elm = getHost(sr);
+        const wrappedListener = getWrappedShadowRootListener(sr, listener);
+        attachDOMListener(elm, type, wrappedListener);
+    }
 }
 
 export function removeShadowRootEventListener(
     sr: SyntheticShadowRootInterface,
     type: string,
-    listener: EventListener,
+    listener: EventListenerOrEventListenerObject,
     _options?: boolean | AddEventListenerOptions
 ) {
-    const elm = getHost(sr);
-    const wrappedListener = getWrappedShadowRootListener(sr, listener);
-    detachDOMListener(elm, type, wrappedListener);
+    // TODO [#1824]: Lift this restriction on the option parameter
+    if (isFunction(listener)) {
+        const elm = getHost(sr);
+        const wrappedListener = getWrappedShadowRootListener(sr, listener);
```

Comment:
You should also remove the original listener, to account for usecases where the listener was added before synthetic-shadow was initialized on the page(like lightning out)

Reply:
Thanks @ravijayaramappa, I added back the line to remove the unwrapped listener for the `window-event-target` polyfill [here](https://github.com/salesforce/lwc/pull/2117/commits/f3522121123d8e8c775d96144353d25494474849) but I don't think we need to do that in this particular method because it's impossible to add listeners to the synthetic `shadowRoot` until this "polyfill" is applied.